app/src/main/java/com/simply/birthdayapp/main/addEvent/data/repository/BirthdayRepositoryImpl.kt

```diff
@@ -59,4 +62,72 @@ class BirthdayRepositoryImpl(
     }.catch { e ->
         emit(Result.Error(e.message ?: "", Birthday()))
     }.flowOn(Dispatchers.IO)
+
+    override fun updateBirthday(id: Int, input: CreateBirthdayInputDomain): Flow<Result<Birthday>> = flow {
+        emit(Result.Loading(Birthday()))
+        val inputModel = input.toCreateBirthdayInputData()
+        val response = apolloClient.mutation(
+            UpdateBirthdayMutation(
+                id = id,
+                UpdateBirthdayInput(
+                    date = Optional.presentIfNotNull(inputModel.date),
+                    image = Optional.presentIfNotNull(inputModel.image),
+                    message = Optional.presentIfNotNull(inputModel.message),
+                    name = Optional.presentIfNotNull(inputModel.name),
+                    relation = Optional.presentIfNotNull(inputModel.relation)
+                )
+            )
+        ).execute()
+        val birthdayData = response.data?.updateBirthday
+        if (birthdayData != null) {
+            val birthday = BirthdayDataModel(
+                createdAt = birthdayData.createdAt.toString(),
```

Comment:
I wouldn't make the mapping here, instead I would create a mapper from BirthdayData to Birthday